Improve Contribution Guide

hendhamdi opened the pull request:
Hello @maintainer, I've added a detailed contribution guide to help new contributors. Here's what I included:
- Instructions for setting up the development environment.
- Steps for submitting a PR.
- Best practices for contributions.

I hope this meets your expectations. Please let me know if any changes are needed. Thank you! 😊

H1manshu-Kumar commented on CONTRIBUTING.md:
````diff
@@ -1,19 +1,66 @@
+
+5. **Push Your Changes**:
+    ```bash
+    Copy
````
Why did you include 'Copy' in the **Push Your Changes** step?

hendhamdi replied:
I'm sorry for the mistake, I will correct it promptly, and thank you for your valuable feedback.

sammyhori commented on CONTRIBUTING.md:
````diff
@@ -1,19 +1,65 @@
+1. **Fork the Project**: Click the "Fork" button at the top right of the repository page to create a copy of the project in your GitHub account.
+
+2. **Clone Your Fork**:
+   ```bash
+   git clone https://github.com/your-username/awesome-for-beginners.git
+   cd awesome-for-beginners
````
For future reference, it is good practice to close your code blocks in markdown (using ```).

hendhamdi replied:
Thank you for the tip! As a beginner, I'm still learning Markdown syntax, and I appreciate your patience. I've fixed the code blocks by properly closing them with triple backticks (```).

sammyhori reviewed (changes requested):
Thanks for the contribution.

Unfortunately I feel this contributing guide is not quite right for this repository.

Firstly, it does not include many of the previous _guidelines_, which state many of the repository specific steps that should be taken and requirements to be followed when considering a contribution on this repository.

Secondly, the guide is too complex, pursing a command line interface approach to cloning and opening a pull request: for those who need a guide for opening pull requests something more simple would be beneficial. On GitHub a user could reasonably easily contribute without leaving the browser, by going on the `data.json` file, pressing the edit button in the top right of the file, copying one of the yaml entries and filling it in with their own information. Then GitHub will guide them through making a pull request using their edit. We could even point people towards the [first-contributions](https://github.com/firstcontributions/first-contributions) repo for a more comprehensive guide on contributing to a repository.

I do, however, like the idea of your issues section at the bottom, suggesting people open issues if they're not sure what to do or have a suggestion which they're not sure how to implement is a nice idea.

hendhamdi commented:
Hello @maintainer,

Thank you for your feedback! I've updated the CONTRIBUTING.md file to better align with the repository's needs. As a beginner, I really appreciate your guidance, and I've made the following changes:

Simplified the Contribution Process: The guide now focuses on contributing directly via GitHub's web interface, making it easier for beginners. I also added a direct link to the data.json file for quick access.

Kept the "Reporting an Issue" Section: I retained and improved this section, as you mentioned it was a good idea.

Added Additional Resources: I included a link to the First Contributions repository for users who want a more comprehensive guide.

Please let me know if there are any further adjustments needed. I'm happy to make additional changes to ensure the guide meets the repository's standards.

Thank you again for your patience and support! 😊

sammyhori commented:
Parts of this was implemented into a final solution in #1604, thanks a lot for your contribution :)
